.azure-pipelines/azure-pipelines-UpgrateVersion.yml

```diff
@@ -18,10 +18,25 @@ pool: sonicbld
 
 stages:
 - stage: Build
+  variables:
+    CACHE_MODE: none
+    VERSION_CONTROL_OPTIONS: 'SONIC_VERSION_CONTROL_COMPONENTS='
   jobs:
   - template: azure-pipelines-build.yml
     parameters:
-      buildSlave: y
+      buildOptions: '${{ variables.VERSION_CONTROL_OPTIONS }} SONIC_BUILD_JOBS=$(nproc) ENABLE_IMAGE_SIGNATURE=y'
+      preSteps:
+      - script: |
+          containers=$(docker container ls | grep "sonic-slave" | awk '{ print $1 }')
+          if [ ! -z "$containers" ]; then
+            docker container kill $containers || true
+            sleep 5
+          fi
+          images=$(docker images 'sonic-slave-*' -a -q)
+          if [ ! -z "$images" ]; then
+            docker rmi -f $images
```

Comment:
>docker rmi -f $images [](http://example.com/codeflow?start=12&length=21)

Also very dangerous. Only `docker rmi` you are responsible. For example, if the image is created by your job, you can remove it.

Reply:
The image is created by last build job. We need to clean them to cover whole build process. It is by design.